pkg/controller/utils/errorscrub.go

```diff
@@ -8,7 +8,8 @@ var (
 	newlineTabRE           = regexp.MustCompile(`\n\t`)
 	certificateTimeErrorRE = regexp.MustCompile(`: current time \S+ is after \S+`)
 	// aws
-	awsRequestIDRE = regexp.MustCompile(`(, )*(?i)(request id: )(?:[-[:xdigit:]]+)`)
+	awsRequestIDRE   = regexp.MustCompile(`(, )*(?i)(request id: )(?:[-[:xdigit:]]+)`)
+	awsNotAuthorized = regexp.MustCompile(`User: \S+ is not authorized to perform: \S+ on resource: \S+`)
```

Comment:
If we wanted to preserve a bit more information here, we could do something like
```suggestion
	awsNotAuthorized = regexp.MustCompile(`(User:) \S+ (is not authorized to perform: \S+ on resource:) \S+`)
```
...and then on L25:

```go
	s = awsNotAuthorized.ReplaceAllString(s, "$1 XXX $2 XXX;")
```

Resulting in: `"failed to describe load balancer for the cluster: AccessDenied: User: XXX is not authorized to perform: sts:AssumeRole on resource: XXX; status code: 403"`

Or if we're confident in the format of the user/resource, even:
```suggestion
	awsNotAuthorized = regexp.MustCompile(`(User: arn:aws:sts::)[^:]+(:assumed-role/[^/]+/)\S+( is not authorized to perform: \S+ on resource: arn:aws:iam::)[^:]+(:\S+)`)
```
and
```go
	s = awsNotAuthorized.ReplaceAllString(s, "${1}XXX${2}XXX${3}XXX${4}")
```
resulting in `"failed to describe load balancer for the cluster: AccessDenied: User: arn:aws:sts::XXX:assumed-role/RH-Managed-OpenShift-Installer/XXX is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::XXX:role/ManagedOpenShift-Installer-Role, status code: 403"`

Reply:
I like the second approach where we replace the identifiers. I've committed that with a few changes. I'm not sure how to remove the extra spaces that get inserted in the replace though.

Reply:
See below